### DISCUSSION TITLE
kotlinx challenge

### DISCUSSION THREAD

**Discussion by t3kh9h** [tyron12233/CodeAssist#438] [Q&A]
I'm trying to use kotlinx libraries and I don't know how to add kotlinx to the library dependencies.
This is what I have so far:

`package ca.t3kh9h.playground.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Job
import kotlin.coroutines.CoroutineContext

abstract class BaseViewModel<T>(protectd val uiContext: CoroutineContext) : ViewModel(), CoroutineScope {
	abstract fun handleEvent(event: T)
	
	//cancellation
	protected lateinit var jobTracker: Job
	
	init {
		jobTracker = Job()
	}
	
	//suggestion from Al Warren: to promote encapsulation and immutability, 
	//hide the MutableLiveData objects behind LiveData references:
	protected val errorState = MutableLiveData<String>()
	val error: LiveData<String> get() = errorState
	
	protected val loadingState = MutableLiveData<Unit>()
	val loading: LiveData<Unit> get() = loadingState
	
	override val coroutineContext: CoroutineContext
		get() = uiContext + jobTracker
}`

![image](https://user-images.githubusercontent.com/105921656/169516582-60f8aa65-5b52-449f-8b5c-f866fe6b5216.png)
![image](https://user-images.githubusercontent.com/105921656/169516697-8fd8f7e8-6619-442a-96f9-9ed1cfdf77c1.png)
The last line in Library Dependencies is wrong, don't know how to remove it.

**Comment by judeetuka** [comment]
Try ;
Open the app and select your desired project. Under the app directory you'll find libraries.json file. Open it and just remove the library from the.

OR

In the library manager, long press the library you want too remove then press delete
